src/main/core/Controller/APINew/Workspace/RegistrationController.php

```diff
@@ -249,66 +255,26 @@ public function unregisterGroupsAction(Request $request, Workspace $workspace):
 
     /**
      * @ApiDoc(
-     *     description="Register users in different workspaces.",
-     *     queryString={
-     *         {"name": "users", "type": "array", "description": "The list of user uuids."},
-     *         {"name": "workspaces", "type": "array", "description": "The list of workspace uuids."},
-     *     },
+     *     description="Register users/groups in different workspaces.",
      *     parameters={
-     *         {"name": "role", "type": {"string"}, "description": "The role translation key"}
+     *         {"name": "role", "type": {"string"}, "description": "The role translation key"},
      *     }
      * )
      *
      * @Route(
-     *    "/users/register/{role}",
-     *    name="apiv2_workspace_bulk_register_users",
+     *    "/register/{role}",
+     *    name="apiv2_workspace_register",
      *    methods={"PATCH"},
      *    defaults={"role":""}
      * )
      */
-    public function bulkRegisterUsersAction(string $role, Request $request): JsonResponse
+    public function registerAction(string $role, Request $request): JsonResponse
     {
-        $workspaces = $this->decodeIdsString($request, Workspace::class, 'workspaces');
-        $users = $this->decodeIdsString($request, User::class, 'users');
-
-        foreach ($workspaces as $workspace) {
-            if ('' === $role) {
-                $roleEntity = $workspace->getDefaultRole();
-            } else {
-                $roleEntity = $this->om->getRepository(Role::class)
-                    ->findOneBy(['translationKey' => $role, 'workspace' => $workspace]);
-            }
-            $this->crud->patch($roleEntity, 'user', Crud::COLLECTION_ADD, $users);
-        }
-
-        return new JsonResponse(array_map(function ($workspace) {
-            return $this->serializer->serialize($workspace);
-        }, $workspaces));
-    }
+        $data = json_decode($request->getContent(), true);
```

Comment:
$this->decodeRequest au lieu de le faire à la main (fourni par le trait RequestDecoder)